pingora-core/src/protocols/http/v1/server.rs

```diff
@@ -609,6 +609,11 @@ impl HttpSession {
         Ok(res)
     }
 
+    /// Return how many request header and body bytes that have been read
+    pub fn request_length(&self) -> usize {
+        self.buf.len()
```

Comment:
I believe this is the `buf` populated [while parsing the request header](https://github.com/cloudflare/pingora/blob/main/pingora-core/src/protocols/http/v1/server.rs#L219), and doesn't increment for body bytes beyond any sent together with the header. We should try to expose / grab whatever the byte count from the `body_reader` as well and combine the counts.

Reply:
While we're at it, it would be nice to have something like `body_bytes_read` for h1/h2 as well to expose just response body bytes.